packages/@aws-cdk/aws-apigatewayv2/lib/http/stage.ts

```diff
@@ -163,20 +175,71 @@ export class HttpStage extends HttpStageBase {
       physicalName: props.stageName ? props.stageName : DEFAULT_STAGE_NAME,
     });
 
+    this.api = props.httpApi;
+    this.baseApi = props.httpApi;
+    this.stageName = this.physicalName;
+
+    const rateLimits = !props.throttle ? undefined : {
+      throttlingBurstLimit: !props.throttle.burstLimit ? undefined : props.throttle.burstLimit,
+      throttlingRateLimit: !props.throttle.rateLimit ? undefined : props.throttle.rateLimit,
+    };
+
+    const detailedMetrics = !props.detailedMetricsEnabled ? undefined : {
+      detailedMetricsEnabled: props.detailedMetricsEnabled,
+    };
+
+    let destinationArn: string | undefined = undefined;
+    if (props.accessLogEnabled) {
+      if (!props.accessLogGroupArn) {
+        // We need to set up the right permissions to create the log group.
+        const iamRoleForLogGroup = new Role(this, 'IAMRoleForAccessLog', {
+          roleName: PhysicalName.GENERATE_IF_NEEDED,
+          assumedBy: new ServicePrincipal('apigateway.amazonaws.com'),
+        });
+
+        iamRoleForLogGroup.node.addDependency(this.api);
+        iamRoleForLogGroup.applyRemovalPolicy(RemovalPolicy.DESTROY);
```

Comment:
We should actually be explicitly setting this to `RETAIN`  since this role is used with the `CfnAccount`. There can only be a single `CfnAccount` per environment (even if there are multiple api gateways). We need to ensure that if we delete the stack these resources remain so they do not break other gateways that are using them.

Reply:
Done